### ISSUE TITLE
No success with the new InstanceIdentifier and .contentVisibility settings

### ISSUE THREAD

**Issue by perez987** [acidanthera/bugtracker#2307]
Maybe it's my mistake and I haven't quite understood the instructions of the Configuration.pdf file, if it's the case, sorry to post in the bug tracker.
However, I think I have followed the instructions and I can't get this new feature to work.

I'm on a Z390 mobo with 6600 XT.
I've plugged 3 disks, macOS13R (Ventura, SATA), macOS14 (Sonoma, SATA) and macOS13 (Ventura, USB).
 
**macOS13R**

- `InstanceIdentifier=macOS13R`
- .contentVisibility: `macOS13,macOS14:Disabled` (no whitespaces)
- Result: I see the 3 disks plus 1 EFI (I should see only macOS13R).
![macOS13R](https://github.com/acidanthera/bugtracker/assets/50429399/5842a7b4-7e3b-4c4c-97c1-f3998743baa8)

**macOS14**

- `InstanceIdentifier=macOS14`
- .contentVisibility: `macOS13,macOS13R:Auxiliary` (no whitespaces)
- Result: I see the 3 disks plus 2 EFI (I should see only macOS14 and the other 2 disks after pressing space bar).
![macOS14](https://github.com/acidanthera/bugtracker/assets/50429399/5d06ae64-4cbd-43a1-a32a-fd8d6b78415e)

**macOS13**

- `InstanceIdentifier=macOS13`
- .contentVisibility: `macOS13R:Disabled` (no whitespaces)
- Result: I see the 3 disks plus 1 EFI (I should see only macOS13 and macOS14).
![macOS13](https://github.com/acidanthera/bugtracker/assets/50429399/94264513-f77e-48f5-a7be-5bed8cd99124)

In summary, I haven't success. Thanks.

**Comment by mikebeaton** [comment]
Hi,

The instance identifiers refer to instances of OpenCore, and the contentVisbility instructions, for example placed next to boot.efi of a given macOS, state which instances of OpenCore can boot it (more precisely, they state which instances of OpenCore should display it, the rest is up to the user).

It is probably worth clarifying, we believe that a correctly tuned config.plist can successfully boot all versions of macOS, so using this feature is not required; it was provided as a convenience, for the setup which some users have found easier to configure or maintain, of different instances of OpenCore for different OS (typically, but not necessarily, different macOS). Again, it isn't definitely isn't recommended, or considered required, to use multiple instances of OC for this reason, but it is now supported.

**Comment by perez987** [comment]
@mikebeaton 
Thank you for the explanation.

**Comment by mikebeaton** [comment]
@5T33Z0 - No major jokes, I hope. Was unclear?

**Comment by perez987** [comment]
@mikebeaton 
Good night.
I know it's an invalid topic.
Would you be so kind to show me your setup, e.g. 2 InstanceIdentifier and its corresponding .contentVisibility files? Because no matter how hard I try to test this feature by strictly (in my opinion) following the instructions, I can't get it to work.
Thank you so much.

**Comment by mikebeaton** [comment]
Hi - that won't be much help to anybody else, will it?

If you still believe that's an issue, please provide your own config.plist (or at least InstanceIdentifier value) from one of more instances of OpenCore, the contents of one or more .contentVisibility files, and where you have placed them (i.e. file path to them) and a one or more debug logs, from the same instances of OpenCore for which you provided the config.plist (or at least InstanceIdentifier).

**Comment by perez987** [comment]
Ok, I have got all data although I'm not sure yet if it's an OC issue or a mistake of mine. Anyway, thanks for your attention.
Tried with 2 OpenCore on 2 different disks.

**OpenCore 1**

- macOS 14 Sonoma b2
- `InstanceIdentifier=Sonoma`
- .contentVisibility next to BOOTx64.efi or next to OpenCore.efi or in both sides (tried 3 ways)
- .contentVisibility content -> `Ventura:Auxiliary`
- To have OpenCore 2 (`InstanceIdentifier=Ventura`) as Auxiliary
- But both Opencore are shown, plus an EFI
- Pressing space bar no more EFIs are shown
- Log -> opencore-Sonoma.txt.

**OpenCore 2**

- macOS 13 Ventura 13.4.1
- `InstanceIdentifier=Ventura`
- .contentVisibility next to BOOTx64.efi or next to OpenCore.efi or in both sides (tried 3 ways)
- .contentVisibility content -> `Sonoma:Disabled`
- To have OpenCore 1 (`InstanceIdentifier=Sonoma`) as Disabled
- But both Opencore are shown plus an EFI
- Pressing space bar one more EFI is shown
- Log - > opencore-Ventura.txt.


[opencore-Sonoma.txt](https://github.com/acidanthera/bugtracker/files/11900330/opencore-Sonoma.txt)
[opencore-Ventura.txt](https://github.com/acidanthera/bugtracker/files/11900331/opencore-Ventura.txt)

**Comment by mikebeaton** [comment]
.contentVisibility needs to be next to the item which will be booted. This is confusing because .contentVisibility was previously only normally placed next to OpenCore.efi and BOOTx64.efi - precisely to stop these being displayed as boot items by other instances of OpenCore. To prevent macOS itself showing as a boot item to an instance of OpenCore, you have to place your instance specific .contentVisibility file next to /System/Volumes/Preboot/{UUID}/System/Library/CoreServices/boot.efi (SIP must be terribly disabled, in order to create this file).
